services/121-service/src/debit-cards-intersolve-visa/debit-cards-intersolve-visa.service.ts

```diff
@@ -120,87 +114,52 @@ export class DebitCardsIntersolveVisaService {
       );
     }
 
-    const intersolveVisaConfig =
-      await this.programFspConfigurationRepository.getPropertiesByNamesOrThrow({
-        programFspConfigurationId: registration.programFspConfigurationId,
-        names: [
-          FspConfigurationProperties.brandCode,
-          FspConfigurationProperties.coverLetterCode,
-        ],
-      });
+    await this.replaceCard(registration, tokenCode);
 
-    //  TODO: REFACTOR: This 'ugly' code is now also in payments.service.createAndAddIntersolveVisaTransactionJobs. This should be refactored when there's a better way of getting registration data.
-    const intersolveVisaAttributes =
-      FSP_SETTINGS[Fsps.intersolveVisa].attributes;
+    return registration;
+  }
 
-    const intersolveVisaAttributeNames = intersolveVisaAttributes.map(
-      (q) => q.name,
-    );
-    const dataFieldNames = [
-      FspAttributes.fullName,
-      FspAttributes.phoneNumber,
-      ...intersolveVisaAttributeNames,
-    ];
+  public async replaceCard(
+    registration: RegistrationEntity,
+    tokenCode?: string,
+  ): Promise<void> {
+    if (tokenCode) {
+      await this.checkIfCardIsAlreadyLinked(tokenCode);
+    }
 
-    const registrationData =
-      await this.registrationDataScopedRepository.getRegistrationDataArrayByName(
-        registration,
-        dataFieldNames,
-      );
+    const contactInfo: DebitCardsContactInfo =
+      await this.getContactInformation(registration);
 
-    if (!registrationData || registrationData.length === 0) {
+    if (this.validateContactInfo(contactInfo)) {
+      await this.sendCustomerInformationToIntersolve({
+        registration,
+        contactInfo,
+      });
+    } else {
       throw new HttpException(
-        `No registration data found for referenceId: ${referenceId}`,
-        HttpStatus.NOT_FOUND,
+        `Fields are missing in contact informatoion`,
+        HttpStatus.BAD_REQUEST,
       );
     }
 
-    const mappedRegistrationData = registrationData.reduce(
-      (acc, { name, value }) => {
-        acc[name] = value;
-        return acc;
-      },
-      {},
+    const intersolveVisaConfig = await this.getIntersolveVisaConfig(
+      registration.programFspConfigurationId,
     );
-
-    for (const name of dataFieldNames) {
-      if (name === FspAttributes.addressHouseNumberAddition) continue; // Skip non-required property
-      if (
-        mappedRegistrationData[name] === null ||
-        mappedRegistrationData[name] === undefined ||
-        mappedRegistrationData[name] === ''
-      ) {
-        throw new HttpException(
-          `Property ${name} is undefined`,
-          HttpStatus.BAD_REQUEST,
-        );
-      }
-    }
-    await this.sendCustomerInformationToIntersolve(registration);
+    const brandCode = intersolveVisaConfig.get(
+      FspConfigurationProperties.brandCode,
+    ) as string;
+    const coverLetterCode = intersolveVisaConfig.get(
+      FspConfigurationProperties.coverLetterCode,
+    ) as string;
```

Comment:
I would use ProgramFspConfigurationRepository.getPropertyValueByName to get brandcode and coverLetter code here